contrib/cmd/recvtty/recvtty.go

```diff
@@ -90,42 +90,53 @@ func handleSingle(path string, noStdin bool) error {
 		return fmt.Errorf("failed to cast to unixconn")
 	}
 
-	socket, err := unixconn.File()
-	if err != nil {
-		return err
+	socket, retErr := unixconn.File()
+	if retErr != nil {
+		return
 	}
 	defer socket.Close()
 
 	// Get the master file descriptor from runC.
-	master, err := utils.RecvFd(socket)
-	if err != nil {
-		return err
+	master, retErr := utils.RecvFd(socket)
+	if retErr != nil {
+		return
 	}
-	c, err := console.ConsoleFromFile(master)
-	if err != nil {
-		return err
+	c, retErr := console.ConsoleFromFile(master)
+	if retErr != nil {
+		return
 	}
 	if err := console.ClearONLCR(c.Fd()); err != nil {
 		return err
 	}
 
 	// Copy from our stdio to the master fd.
 	quitChan := make(chan struct{})
+	errChan := make(chan error)
 	go func() {
-		io.Copy(os.Stdout, c)
+		_, err := io.Copy(os.Stdout, c)
+		if err != nil {
+			errChan <- err
+		}
 		quitChan <- struct{}{}
```

Comment:
IOW this could be simple
```go
_, err := io.Copy(os.Stdout, c)
chan <- err
```

and the code that's receiving this will check for nil